examples/ffmpeg_vad_streaming/index.js

```diff
@@ -59,59 +60,74 @@ const model_load_end = process.hrtime(model_load_start);
+  const lm_load_end = process.hrtime(lm_load_start);
+  console.error('Loaded language model in %ds.', totalTime(lm_load_end));
 }
 
-const vad = new VAD(VAD.Mode.NORMAL);
-const voice = {START: true, STOP: false};
-let sctx = model.setupStream(150, 16000);
-let state = voice.STOP;
+// Default initial allocation = 3 seconds := 150
+const PRE_ALLOC_FRAMES = 150;
+
+// Default is 16kHz
+const AUDIO_SAMPLE_RATE = 16000;
+
+const VAD_MODE = VAD.Mode.NORMAL;
+// const VAD_MODE = VAD.Mode.LOW_BITRATE;
+// const VAD_MODE = VAD.Mode.AGGRESSIVE;
+// const VAD_MODE = VAD.Mode.VERY_AGGRESSIVE;
+
+// Debounce time
+const DEBOUNCE_TIME = 20;
+
+// Modes are NORMAL, LOW_BITRATE, AGGRESSIVE, VERY_AGGRESSIVE
+const VAD_STREAM = VAD.createStream({
+  mode: VAD_MODE,
+  audioFrequency: AUDIO_SAMPLE_RATE,
+  debounceTime: DEBOUNCE_TIME
+});
+
+// Spawn ffmpeg process
+const ffmpeg = spawn('ffmpeg', [
+  '-hide_banner',
+  '-nostats',
+  '-loglevel', 'fatal',
+  '-i', args['audio'],
+  '-vn',
+  '-acodec', 'pcm_s16le',
+  '-ac', 1,
+  '-ar', AUDIO_SAMPLE_RATE,
+  '-f', 's16le',
+  'pipe:'
+]);
+
+let audioLength = 0;
+let sctx = model.setupStream(PRE_ALLOC_FRAMES, AUDIO_SAMPLE_RATE);
 
 function finishStream() {
-	const model_load_start = process.hrtime();
-	console.error('Running inference.');
-	console.log('Transcription: ', model.finishStream(sctx));
-	const model_load_end = process.hrtime(model_load_start);
-	console.error('Inference took %ds.', totalTime(model_load_end));
+  const model_load_start = process.hrtime();
+  console.error('Running inference.');
+  console.log('Transcription: ', model.finishStream(sctx));
+  const model_load_end = process.hrtime(model_load_start);
+  console.error('Inference took %ds for %ds audio file.', totalTime(model_load_end), audioLength.toPrecision(4));
+  audioLength = 0;
 }
 
-let ffmpeg = require('child_process').spawn('ffmpeg', [
-	'-hide_banner',
-	'-nostats',
-	'-loglevel', 'fatal',
-	'-i', args['audio'],
-	'-af', 'highpass=f=200,lowpass=f=3000',
-	'-vn',
-	'-acodec', 'pcm_s16le',
-	'-ac', 1,
-	'-ar', 16000,
-	'-f', 's16le',
-	'pipe:'
-]);
+function intermediateDecode() {
+  finishStream();
+  sctx = model.setupStream(PRE_ALLOC_FRAMES, AUDIO_SAMPLE_RATE);
+}
 
-ffmpeg.stdout.on('data', chunk => {
-	vad.processAudio(chunk, 16000).then(res => {
-		switch (res) {
-			case VAD.Event.SILENCE:
-				if (state === voice.START) {
-					state = voice.STOP;
-					finishStream();
-					sctx = model.setupStream(150,16000);
-				}
-				break;
-			case VAD.Event.VOICE:
-				state = voice.START;
-				model.feedAudioContent(sctx, chunk.slice(0, chunk.length / 2));
-				break;
-		}
-	});
-});
+function feedAudioContent(chunk) {
+  audioLength += (chunk.length / 2) * ( 1 / AUDIO_SAMPLE_RATE);
+  model.feedAudioContent(sctx, chunk.slice(0, chunk.length / 2));
+}
 
-ffmpeg.stdout.on('close', code => {
-	finishStream();
-});
+function processVad(data) {
+  if (data.speech.start||data.speech.state) feedAudioContent(data.audioData)
```

Comment:
Please retrain on single line without `{}`, it's a time bomb. Also space around ` || `